### PR TITLE
module: make require.main identical to main module

### PR DESCRIPTION
**Pull request by Txiaozhe**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added

### CODE REVIEW

**Comment by Txiaozhe** [comment]
#448

**Comment by ErosZy** [comment]
Maybe need more tests. In node v8.x doc described:

`
For a file foo.js, this will be true if run via node foo.js, but false if run by require('./foo').
`
But we only test first situation. `require('foo.js')` not tests :)

**Comment by yorkie** [comment]
+1 on @ErosZy's suggestion :)

**Comment by Txiaozhe** [comment]
Yes, It is rigorous, thanks @ErosZy, I will try it.

**Comment by yorkie** [comment]
BTW, I think we could push this to upstream(IoT.js).

**Comment by Txiaozhe** [comment]
> BTW, I think we could push this to upstream(IoT.js).

Good idea!

File: test/run_pass/test_require_main_foo.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+'use strict';
```
Comment (legendecas) [inline test/run_pass/test_require_main_foo.js]: This test is not included in the test sets :(. 

As a suggestion, the file could be put in fixtures folder and then being required to cover the false case.

Comment (Txiaozhe) [reply test/run_pass/test_require_main_foo.js]: This only be invoked by `test_require_main.js`, but I have not found the `fixtures` folder, will I create one?

Comment (legendecas) [reply test/run_pass/test_require_main_foo.js]: Samples could be found: `test/run_pass/require1`, `test/run_pass/require2`, `test/run_pass/require3`

Comment (Txiaozhe) [reply test/run_pass/test_require_main_foo.js]: Yes, it's concise and clear, I will do it.

File: src/js/module.js [hunk 2]
```diff
@@ -250,6 +250,10 @@ iotjs_module_t.load = function(id, parent) {
   iotjs_module_t.cache[modPath] = module;
   iotjs_module_t.curr = modPath;
 
+  if (isMain) {
+    mainModule = module;
```
Comment (yorkie) [inline src/js/module.js]: Miss https://github.com/nodejs/node/blob/master/lib/internal/modules/cjs/loader.js#L547

File: src/js/module.js [hunk 3]
```diff
@@ -340,8 +344,8 @@ iotjs_module_t.runMain = function() {
     var fn = process.debuggerSourceCompile();
     fn.call();
   } else {
-    var filename = mainModule.filename = process.argv[1];
-    mainModule.exports = iotjs_module_t.load(filename, null);
+    var filename = process.argv[1];
+    iotjs_module_t.load(filename, null, true);
```
Comment (yorkie) [inline src/js/module.js]:
```js
iotjs_module_t.load(process.argv[1], null, true);
```

File: test/run_pass/require_main/foo.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+'use strict';
+
+module.exports = function() {
+  return require.main === module;
+}
```
Comment (yorkie) [inline test/run_pass/require_main/foo.js]: This file is not testing anything, move this to fixtures directory.

Comment (Txiaozhe) [reply test/run_pass/require_main/foo.js]: But I have move it into `run_pass/require_main` dir, it is fault?

Comment (yorkie) [reply test/run_pass/require_main/foo.js]: Misunderstood, but a suggestion is the name to be require4 or add your test to exists require dir.

Comment (Txiaozhe) [reply test/run_pass/require_main/foo.js]: Maybe it is more appropriate, I will do it.

Comment (yorkie) [reply test/run_pass/require_main/foo.js]: Thanks, the Node.js team maintains a huge tests now, ShadowNode doesn't want to do like that, thus we should own a better group for our tests.

Comment (Txiaozhe) [reply test/run_pass/require_main/foo.js]: yeah, it is necessary！:)

File: test/run_pass/require4/foo.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+'use strict';
+
+module.exports = function() {
+  return require.main === module;
+}
```
Comment (yorkie) [inline test/run_pass/require4/foo.js]: Missing the last `;`.

Comment (Txiaozhe) [reply test/run_pass/require4/foo.js]: Yeah, But out lint not scentout.

**Pull request merged**